repository.go

```diff
@@ -274,6 +318,53 @@ func decodeRepositoryFiles(repoResponse interface{}) ([]RepositoryFile, error) {
 	return *repositoryFiles, nil
 }
 
+func decodeRepositoryBranches(branchResponse interface{}) (*RepositoryBranches, error) {
+
+	var branchResponseMap map[string]interface{}
+	err := json.Unmarshal(branchResponse.([]byte), &branchResponseMap)
+	if err != nil {
+		return nil, err
+	}
+
+	branchArray := branchResponseMap["values"].([]interface{})
+	var branches []RepositoryBranch
+	for _, branchEntry := range branchArray {
+		var branch RepositoryBranch
+		err = mapstructure.Decode(branchEntry, &branch)
+		if err == nil {
+			branches = append(branches, branch)
+		}
+	}
+
+	page, ok := branchResponseMap["page"].(float64)
+	if !ok {
+		page = 0
+	}
```

Comment:
I'm curious as to your thoughts on this section.  This set of conditionals that evaluate the metadata (`page`, `pagelen`, `size`) in the response from Bitbucket should always be valued; however, if something is missing, the current behavior will default it to a value of `0`.  I'm not certain how consumers would treat or view this response; any insights on a possibly better idea?

Reply:
It is no problem, I think. At the struct definition, `0` is correct If you use int types as `Page int`.
(https://github.com/ktrysmt/go-bitbucket/pull/66/files#diff-ab2c7209d94fb144c8d55246ad821816R47)

So gopher would be care to use to struct and zero types. @mfb2